tuf/encoding/asn1_metadata_definitions.py

```diff
@@ -0,0 +1,205 @@
+"""
+These are the pyasn1-compatible ASN.1 definitions for TUF metadata. Please also
+see tuf_metadata_definitions.asn1
+
+This file is in part automatically generated by asn1ate (v.0.6.0), from
+tuf_metadata_definitions.asn1
+
+
+It is then hand-modified to do the following:
+
+  - remove unneeded subtyping (almost all subtyping)
+
+  - add MAX value
+
+  - when providing a default value, rather than a subtyped object constrained to
+     only have one value, provide a specific value (both seem to work, but I
+     think this has the right semantics)
+     e.g. use:   DefaultedNamedType('terminating', univ.Boolean(0))
+     instead of: DefaultedNamedType('terminating', univ.Boolean().subtype(value=0))
+
+  - Move superclass value overrides into bodies of classes
+     e.g. use:
+       class Positive(univ.Integer):
+         subtypeSpec = constraint.ValueRangeConstraint(1, MAX)
+     instead of:
+       class Positive(univ.Integer):
+         pass
+       Positive.subtypeSpec = constraint.ValueRangeConstraint(1, MAX)
+
+  - changed imports for style and readability
+
+  - tab length 2, not 4 (per PEP 8)
+
+  - re-ordered to better match order in tuf_metadata_definitions.asn1 for
+    readability (but they must still appear such that depended-on classes appear
+    before the classes that use them).
+
+  - This comment is added to the top of the auto-generated file.
+
+"""
+
+import pyasn1.type.univ as univ
+import pyasn1.type.char as char
+import pyasn1.type.constraint as constraint
+import pyasn1.type.namedtype as namedtype
+# Likely imports for later:
+# import pyasn1.type.namedval as namedval
+# import pyasn1.type.tag as tag
+from pyasn1.type.namedtype import NamedType, NamedTypes, DefaultedNamedType
+
+# Maximum integer value when bounding integer values below.
+# It does not seem possible to place a minimum without placing a maximum.
+# For thresholds, versions, timestamps, etc., we want only non-negative
+# integers, so we're stuck setting a maximum (or not placing any constraints
+# at all in the ASN.1 metadata definition itself, would would also be fine).
+MAX = 2**32-1
```

Comment:
Is this inclusive or exclusive?

Reply:
It's inclusive, which is a little odd, but that's why this is -1.